src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -90,53 +142,112 @@ export default class CoExpressionTab extends React.Component<ICoExpressionTabPro
                     return Promise.resolve([]);
                 }
             }
         }),
-        q=>`${q.entrezGeneId},${q.molecularProfile.molecularProfileId}`
+        q=>`${q.geneticEntityId},${q.subjectProfile.molecularProfileId},${q.queryProfile.molecularProfileId}`
     );
 
-    private get dataSetSelector() {
-        if (this.selectedMolecularProfile &&
+    private get profilesSelector() {
+        if (this.selectedSubjectProfileGene &&
+            this.selectedSubjectProfileGeneset &&
+            this.selectedQueryProfile &&
+            this.selectedGeneticEntityId &&
             this.props.store.molecularProfileIdToProfiledSampleCount.isComplete &&
-            this.props.store.coexpressionTabMolecularProfiles.isComplete) {
+            this.props.store.coexpressionTabMolecularSubjectProfilesGene.isComplete &&
+            this.props.store.coexpressionTabMolecularSubjectProfilesGeneset.isComplete &&
+            this.props.store.coexpressionTabMolecularQueryProfiles.isComplete &&
+            this.props.store.coexpressionTabMolecularQueryProfiles.isComplete) {
+            const subjectOptionsGene = this.props.store.coexpressionTabMolecularSubjectProfilesGene.result.map(profile=>{
+                const profiledSampleCount = this.props.store.molecularProfileIdToProfiledSampleCount.result![profile.molecularProfileId];
+                return {
+                    label: `${profile.name} (${profiledSampleCount} sample${profiledSampleCount !== 1 ? "s" : ""})`,
+                    value: profile.molecularProfileId
+                };
+            });
+            const subjectOptionsGeneset = this.props.store.coexpressionTabMolecularSubjectProfilesGeneset.result.map(profile=>{
+                const profiledSampleCount = this.props.store.molecularProfileIdToProfiledSampleCount.result![profile.molecularProfileId];
+                return {
+                    label: `${profile.name} (${profiledSampleCount} sample${profiledSampleCount !== 1 ? "s" : ""})`,
+                    value: profile.molecularProfileId
+                };
+            });
+            const queryOptions = this.props.store.coexpressionTabMolecularQueryProfiles.result.map(profile=>{
+                    const profiledSampleCount = this.props.store.molecularProfileIdToProfiledSampleCount.result![profile.molecularProfileId];
+                    return {
+                        label: `${profile.name} (${profiledSampleCount} sample${profiledSampleCount !== 1 ? "s" : ""})`,
+                        value: profile.molecularProfileId
+                    };
+                });
             return (
-                <div
+                <div>
+                    <div
+                        style = {{
+                            display: "flex",
+                            alignItems: "center",
+                            padding: "5px"
+                        }}
+                    >
+                        <span>Subject Profile:</span>
```

Comment:
I think `Subject` is a little unclear, should we show this to the user as `${xGeneticEntity.name} Profile` (or y, not sure which it corresponds to)?